fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinReorderLeftDeep.java

```diff
@@ -24,8 +25,35 @@ protected void enumerate() {
             return (diff < 0 ? -1 : (diff > 0 ? 1 : 0));
         });
 
+        boolean useHeuristic = true;
```

Comment:
Seems this var is unnecessary?

Reply:
Yes. I put it here for easier debugging: if you doubt this strategy is better and want to fallback to old strategy.